tests/test_flow/test_reload.py

```diff
@@ -280,3 +281,242 @@ def r():
+# after reloading.
+def test_merge_change_keep_parameter(write_module, flow_tester):
+    write_module(
+        "base_flow_module",
+        """
+            import bionic as bn
+
+            builder = bn.FlowBuilder('base_flow')
+
+            @builder
+            def x():
+                return 1
+
+            @builder
+            def x_plus_1(x):
+                return x + 1
+
+            @builder
+            def total(x):
+                return x
+
+            base_flow = builder.build()
+        """,
+    )
+
+    def write_flow_module(keep):
+        flow_tester.write_flow_module(
+            f"""
+                from base_flow_module import base_flow
+
+                builder = bn.FlowBuilder('flow')
+                builder.set('core__versioning_mode', 'auto')
+
+                @builder
+                def x():
+                    return 2
+
+                @builder
+                def y():
+                    return 20
+
+                @builder
+                def total(x, y):
+                    return x + y
+
+                builder.merge(base_flow, keep='{keep}')
+            """,
+        )
+
+    write_flow_module("new")
+    flow_tester.expect_values(x=1, x_plus_1=2, y=20, total=1)
+
+    write_flow_module("old")
+    flow_tester.reload_flow()
+    flow_tester.expect_values(x=2, x_plus_1=3, y=20, total=22)
+
+    write_flow_module("new")
+    flow_tester.reload_flow()
+    flow_tester.expect_values(x=1, x_plus_1=2, y=20, total=1)
+
+
+def test_change_import_builder(write_module, flow_tester):
+    def write_builder_module(x: int):
+        write_module(
+            "base_builder_module",
+            f"""
+                import bionic as bn
+
+                builder = bn.FlowBuilder('flow')
+
+                @builder
+                def x():
+                    return {x}
+
+                @builder
+                def x_plus_1(x):
+                    return x + 1
+
+                @builder
+                def total(x):
+                    return x
+            """,
+        )
+
+    write_builder_module(1)
+    flow_tester.write_flow_module(
+        """
+            from base_builder_module import builder
+
+            builder.set('core__versioning_mode', 'auto')
+
+            @builder
+            def y():
+                return 10
+
+            @builder
+            def total(x, y):
+                return x + y
+        """,
+    )
+    flow_tester.expect_values(x=1, x_plus_1=2, y=10, total=11)
+
+    write_builder_module(2)
+    flow_tester.reload_flow()
+    flow_tester.expect_values(x=2, x_plus_1=3, y=10, total=12)
+
+    write_builder_module(3)
+    flow_tester.reload_flow()
+    flow_tester.expect_values(x=3, x_plus_1=4, y=10, total=13)
+
+
+def test_internal_modules_not_reloaded(write_module, flow_tester):
+    write_module("test_module", "")
+    flow_tester.write_flow_module(
+        """
+            import sys
```

Comment:
Could we also test a third-party library, like `pandas`?

Reply:
Good idea